website/src/docs/core.md

```diff
@@ -946,6 +946,19 @@ uppy.on('upload-error', (file, error, response) => {
 })
 ```
 
+### `upload-stalled`
+
+Fired when an upload is seemingly stalled. Use this event to display a message on the UI to tell the user they might want to retry the upload.
+
+```js
+uppy.on('upload-stalled', () => {
+  stallWarning.hidden = false
```

Comment:
rather than having pseudo-code, might be nicer to use our notification system as an example.

```suggestion
  uppy.info(`Your upload "${file.meta.name}" has not made any progress for ${timeout} seconds. You may want to retry it.`, 'warning')
```